docs/book/introduction.md

```diff
@@ -4,27 +4,178 @@ description: Welcome to ZenML!
-* It also features **cloud-** and **tool-agnostic** **interfaces/abstractions** such as orchestration systems, artifact repositories, and model deployment platforms that cover the entire ML journey.
-* With these abstractions, it remains **extensible** and **open to customization**. That is also backed by numerous **collaborative features** that allow you to work not just with your team but also with the community.
-* Moreover, this creates a separation between the different aspects of ML workflows, allowing users to focus on creating their own respective solutions in an efficient manner.
-* Overall, it reduces the complexity of your development lifecycle, all the way **from an experimentation environment to a production-grade solution**.
+```bash
+python run.py  # develop your code locally with all your favorite tools
+zenml stack set production
+python run.py  # run on production infrastructure without any code changes
+```
+
+**Pythonic SDK:** ZenML is designed to be as unintrusive as possible. Adding 
+a ZenML `@step` or `@pipeline` decorator to your Python functions is enough to 
+turn your existing code into ZenML pipelines:
+
+```python
+from zenml import pipeline, step
+
+@step
+def step_1() -> str:
+  return "world"
+
+@step
+def step_2(input_one: str, input_two: str) -> None:
+  combined_str = input_one + ' ' + input_two
+  print(combined_str)
+
+@pipeline
+def my_pipeline():
+  output_step_one = step_1()
+  step_2(input_one="hello", input_two=output_step_one)
+
+my_pipeline()
+```
+
+**Reusability:** Defining your code as ZenML steps brings standardization 
+into your organization, allowing you to easily reuse code developed by your 
+colleagues:
+
+```python
+from my_organization.steps import data_loader_step, model_validation_step
+
+@step
+def my_model_trainer_step(data):
+    ...
+
+@pipeline
+def my_pipeline():
+  data = data_loader_step()
+  model = my_model_trainer_step(data)
+  eval_metric = model_validation_step(model)
+```
+{% endtab %}
+
+{% tab title="ZenML for ML Engineers" %}
+
+As an ML Engineer, ZenML empowers you to take ownership of the entire ML 
+lifecycle end-to-end. Adopting ZenML means fewer handover points and more 
+visibility on what is happening in your organization.
+
+**ML Lifecycle Management:** ZenML's abstractions enable you to manage
+sophisticated ML setups with ease. After you define your ML workflows as 
+[Pipelines](./getting-started/core-concepts.md#1-development) and your 
+development, staging, and production infrastructures as 
+[Stacks](./getting-started/core-concepts.md#2-execution), you can move 
+entire ML workflows to different environments in seconds.
+
+```bash
+zenml stack set staging
+python run.py  # test your workflows on staging infrastructure
+zenml stack set prod
+python run.py  # run your workflows in production
+```
+
+**Reproducibility:** ZenML enables you to painlessly reproduce previous results
+by automatically tracking and versioning stacks, pipelines, and output 
```

Comment:
it's implied by 'pipelines', but do we need to mention code tracking / versioning here?

Reply:
Doesn't hurt I guess